Source/Proxy/IPublicCallContext.cs

```diff
@@ -0,0 +1,21 @@
+using System.Reflection;
+
+namespace Moq.Proxy
+{
+	/// <summary>
+	/// Class that represents the method call currently being executed. 
+	/// Allows for custom interception for open generic methods
+	/// </summary>
+	public interface IPublicCallContext
+	{
+		/// <summary>
+		/// Arguments passed into the method
+		/// </summary>
+		object[] Arguments { get; }
+
+		/// <summary>
+		/// Closed generic method currently being executed.
+		/// </summary>
+		MethodInfo Method { get; }
+	}
+}
```

Comment:
1. Regarding `null` arguments, yes, the user can take a look at the `MethodInfo Method { get; }` property, which will have all the generic arguments populated.

2. I agree, better to start off slim with the interface.

3.  Moved to `Moq`

4. Yes agree as per my initial PR comment, I prefer the public one to be `ICallContext` and the internal one something else, some candidates:
 - `IExtendedCallContext`
 - `IInternalCallContext`
 - `IProxyCallContext`

For now, have used `IExtendedCallContext` as per suggestion :-)